### PR TITLE
Add logging

### PR DESCRIPTION
**Pull request by svenvanderburg**
Add logging to the project. 
Replace most print statements with logger.info statements (sometimes logger.warning)
Update logging statements where necessary
Add new logging statements where necessary

I think we log enough for now, if we need more logging for any reason we can always add it.

### CODE REVIEW

File: proof_of_concept/local_workflow_runner.py [hunk 1]
```diff
@@ -177,8 +177,8 @@ def _get_step_inputs(
                     self._this_runner, data_key))
                 logger.info('Metadata: {}'.format(asset.metadata))
             except KeyError:
-                logger.info(f'Job at {self._this_runner} found input'
-                            f'{data_key} not yet available.')
+                logger.warning(f'Job at {self._this_runner} found input'
```
Comment (LourensVeen) [inline proof_of_concept/local_workflow_runner.py]: I think info is more appropriate here, this is normal expected behaviour. It just means that the previous step hasn't finished yet.

File: proof_of_concept/asset_store.py [hunk 2]
```diff
@@ -53,17 +53,19 @@ def retrieve(self, asset_id: str, requester: str
             KeyError: If no asset with the given id is stored here.
 
         """
-        logger.info(f'{self} servicing request from {requester} for data: '
-                    f'{asset_id}, ')
+        logger.info(f'{self}: servicing request from {requester} for data: '
```
Comment (LourensVeen) [inline proof_of_concept/asset_store.py:56]: Can we discuss f-strings and support for older Python versions?

File: proof_of_concept/local_workflow_runner.py (hunk 1 above)
Comment (svenvanderburg) [reply proof_of_concept/local_workflow_runner.py]: Updated

File: proof_of_concept/asset_store.py (hunk 2 above)
Comment (svenvanderburg) [reply proof_of_concept/asset_store.py:56]: We discussed that we will stick to f-strings for now, and therefore not support older python versions.

**Pull request merged**